### PR TITLE
Mccalluc/events revert access control

### PR DESCRIPTION
**Pull request by mccalluc**
This is a subset of #2812... I'm having more trouble adding the access controls than I expected, but this much should be good.

### CODE REVIEW

**Comment by mccalluc** [comment]
Test failure is from cypress on Satori:
```
  1) Satori Visualization Sorts nodes in list graph by name:
     CypressError: Timed out retrying: Expected to find element: '.node.lock-directly', but never found it.
```

File: refinery/core/serializers.py [hunk 1]
```diff
@@ -165,3 +166,43 @@ class Meta:
...
+    )
+
+    user = serializers.SlugRelatedField(
+        read_only=True,
+        slug_field='username'
+    )
+
+    message = _StrObjectField()
+
+    json = _DeserializeJSONField()
```
Comment (scottx611x) [inline refinery/core/serializers.py]: Is the standard [JSONField](http://www.django-rest-framework.org/api-guide/fields/#jsonfield) not sufficient here?

File: refinery/core/serializers.py [hunk 2]
```diff
@@ -165,3 +166,43 @@ class Meta:
...
+        read_only=True,
+        slug_field='uuid'
+    )
+
+    user = serializers.SlugRelatedField(
+        read_only=True,
+        slug_field='username'
+    )
+
+    message = _StrObjectField()
```
Comment (scottx611x) [inline refinery/core/serializers.py]: I think the same effect could be achieved without having to create a custom field with the introduction of an `Event.get_message()` and a [SerializerMethodField](http://www.django-rest-framework.org/api-guide/fields/#serializermethodfield) here

File: refinery/core/test_views.py [hunk 3]
```diff
@@ -934,3 +939,86 @@ def test_get_workflow_graph(self):
             uuid=self.workflow.uuid
         )
         self.assertEqual(get_response.content, self.mock_workflow_graph)
+
+
+class EventApiV2Tests(APIV2TestCase):
+    def setUp(self):
+        create_public_group()
```
Comment (scottx611x) [inline refinery/core/test_views.py]: Not sure if quick test has wide enough adoption to warrant this change

File: refinery/core/test_views.py [hunk 4]
```diff
@@ -934,3 +939,86 @@ def test_get_workflow_graph(self):
...
+        user = User.objects.create_user('testuser')
+        CuserMiddleware.set_user(user)
+
+        create_tool_with_necessary_models("VISUALIZATION")
+        create_tool_with_necessary_models("WORKFLOW")
+
+        events = Event.objects.all()
+        self.assertEqual(len(events), 4)
+
+        messages = [str(_) for _ in events]
```
Comment (scottx611x) [inline refinery/core/test_views.py]: `_` -> `event`?

File: refinery/core/views.py [hunk 5]
```diff
@@ -698,6 +699,14 @@ class NodeViewSet(viewsets.ModelViewSet):
     # permission_classes = (IsAuthenticated,)
 
 
+class EventViewSet(viewsets.ModelViewSet):
+    """API endpoint that allows Events to be viewed"""
+    queryset = Event.objects.all()
+    serializer_class = EventSerializer
+    # Don't think I need a lookup_field?
```
Comment (scottx611x) [inline refinery/core/views.py]: Yeah I think the default is the `Object.pk`, and at the moment I don't think we're trying to have a detail view for `Events`

File: refinery/core/serializers.py (hunk 1 above)
Comment (scottx611x) [reply refinery/core/serializers.py]: You can use `JSONField` while specifying a `source` that points to a method on the `Event model that will spit out `Event.json` as a dict

See an example here: https://github.com/refinery-platform/refinery-platform/blob/develop/refinery/tool_manager/serializers.py#L79

File: refinery/core/test_views.py [hunk 6]
```diff
@@ -934,3 +939,86 @@ def test_get_workflow_graph(self):
...
+        # Necessary if using "--settings=config.settings.quick_test".
+        # TODO: Move to superclass?
+
+        super(EventApiV2Tests, self).setUp(
+            api_base_name="events/",
+            view=EventViewSet.as_view({"get": "list"})
+        )
+
+    def test_get_event_list(self):
+        user = User.objects.create_user('testuser')
```
Comment (jkmarx) [inline refinery/core/test_views.py]: snake_case

File: refinery/core/test_views.py [hunk 7]
```diff
@@ -934,3 +939,86 @@ def test_get_workflow_graph(self):
...
+        display_names = [
+            json.loads(_.json).get('display_name') for _ in events
+        ]
+        date_times = [
+            _.date_time.isoformat().replace('+00:00', 'Z') for _ in events
+        ]
+
+        get_request = self.factory.get(urljoin(self.url_root, '/'))
+        get_response = self.view(get_request).render()
+        # TODO: Why do I need render()?
```
Comment (jkmarx) [inline refinery/core/test_views.py]: In general, can we remove todo questions from the code and add them to the dev meeting agenda?

File: refinery/core/test_views.py [hunk 8]
```diff
@@ -934,3 +939,86 @@ def test_get_workflow_graph(self):
...
+
+        self.maxDiff = None
+        self.assertEqual(
```
Comment (jkmarx) [inline refinery/core/test_views.py]: Can we separate the len check and the content check? The way the test is written it's repetitive to check the length and check if the arrays are equal. For content, it'd be great to break it down a bit more and check for particular fields or focus on particular difference (ex one vis subtype vs one empty subtype). I'm just thinking how would I debug this test if xx changed?

File: refinery/core/views.py (hunk 5 above)
Comment (jkmarx) [reply refinery/core/views.py]: Comment on which events are returned? Per user?

**Comment by scottx611x** [comment]
Access control provided in PR: https://github.com/refinery-platform/refinery-platform/pull/2816

**Pull request closed without merging**